test/testLiveAsTool.js

```diff
@@ -0,0 +1,150 @@
+var assert = require('assert');
+var path = require('path');
+var fse = require('fs-extra');
+var childProcess = require('child_process');
+var temp = require('temp').track();
+
+var GitHub = require('github');
+var github = new GitHub({
+  version: '3.0.0',
+  protocol: 'https',
+  headers: {
+    'user-agent': 'Oghliner',
+  },
+});
+
+var username = process.env.USER, password = process.env.PASS;
+
+// Skip these tests if the USER or PASS environment variables aren't set.
+if (!username || !password) {
+  return;
+}
+
+function createRepo() {
+  return new Promise(function(resolve, reject) {
+    github.repos.create({
+      name: 'test_oghliner_live',
+      auto_init: true,
+    }, function(err) {
+      if (err) {
+        reject(err);
+      } else {
+        resolve();
+      }
+    });
+  });
+}
+
+function deleteRepo() {
+  return new Promise(function(resolve, reject) {
+    github.repos.delete({
+      user: username,
+      repo: 'test_oghliner_live',
+    }, function(err) {
+      if (err) {
+        reject(err);
+      } else {
+        resolve();
+      }
+    });
+  });
+}
+
+function getBranch() {
+  return new Promise(function(resolve, reject) {
+    setTimeout(function() {
+      github.repos.getBranch({
+        user: username,
+        repo: 'test_oghliner_live',
+        branch: 'gh-pages',
+      }, function(err, res) {
+        if (err) {
+          reject(err);
+        } else {
+          resolve(res);
+        }
+      });
+    }, 3000);
+  });
+}
+
+function spawn(command, args) {
+  return new Promise(function(resolve, reject) {
+    var child = childProcess.spawn(command, args);
+
+    child.stdout.on('data', function(chunk) {
+      process.stdout.write(chunk);
+    });
+
+    child.stderr.on('data', function(chunk) {
+      process.stderr.write(chunk);
+    });
+
+    child.on('exit', function(code, signal) {
+      if (code === 0) {
+        resolve(code);
+      } else {
+        reject(code);
+      }
+    });
+
+    child.on('error', function(err) {
+      reject(err);
+    });
+  });
+}
+
+describe('CLI interface, oghliner as a tool', function() {
+  this.timeout(60000);
+
+  var oldWD = process.cwd();
+
+  before(function() {
+    github.authenticate({
+      type: 'basic',
+      username: username,
+      password: password,
+    });
+  });
+
+  beforeEach(function() {
+    process.chdir(temp.mkdirSync('oghliner'));
+
+    process.env.GH_TOKEN = username + ':' + password;
+
+    return deleteRepo();
+  });
+
+  afterEach(function() {
+    process.chdir(oldWD);
+
+    delete process.env['GH_TOKEN'];
+  });
+
+  it('should work', function() {
+    return createRepo()
+    .then(() => spawn('git', ['clone', 'https://' + username + ':' + password + '@github.com/' + username + '/test_oghliner_live']))
```

Comment:
Arrow functions aren't supported in Node 0.12, which we still claim compatibility.